### ISSUE TITLE
Component - Categories

### ISSUE THREAD

**Issue by twblack88** [makerdao/community-portal#19]
For landing pages, categories give the user the overview of everything available in the section. 

[More detail](https://www.figma.com/file/mVblaaFzkqracWycXl9czh/Maker-comm-dev?node-id=355%3A73) note the status message nested per category. 🥇 

Figma of the [work with us](https://www.figma.com/file/mVblaaFzkqracWycXl9czh/Maker-comm-dev?node-id=355%3A73) landing page has the best approximation of placement and content.

**Comment by MaximumCrash** [comment]
@befitsandpiper

Categories, while they look similarly to Chocolate boxes, are one of the only components that break the consistency of the column width. Based on my math (see below this) they are 7.58% wider than the other content provided. (ie. If the content is at 100% width, we can expect Categories to be 107.58% width)

**My Math** (using [this](https://www.figma.com/file/mVblaaFzkqracWycXl9czh/Maker-comm-dev?node-id=398%3A1) as reference)

Page Width - 1200px
Content Max Width - 885px
Category Width (with Horizontal Margin) - 976px

Content Max Width / Page Width = 74.75%
Category Width / Page Width. = 81.33%

74.75% - 81.33% = **7.58%**

There's a chance this will look disgusting on pages with a sidebar, but it is not up to us to decide where they use this component. So we'll do our best.

This component is another case where we must go against the rules of it's design to ensure it can be scripted flexibly in MDX. What I mean is, while the designs provided Categories in examples of 2x2 It should support an infinite amount of containers.

**NOTE: I would recommend using Grid styles for this since they have specific sizes.**

1 Container
- Max Width 480px
- Align Center

2 Containers
- Max Width 480 each
- Align Center

3 Containers
- Max Width 361px each
- Align Center

![Screen Shot 2020-06-19 at 8 38 18 AM](https://user-images.githubusercontent.com/9157341/85133407-9fe09700-b208-11ea-930e-756d4805aa43.png)


4 Containers
- Max Width 480px each
- Align Center

![Screen Shot 2020-06-19 at 8 38 30 AM](https://user-images.githubusercontent.com/9157341/85133416-a40cb480-b208-11ea-9880-fc611295d080.png)


5 - Infinity
Rinse and Repeat 3 Container styles


**Acceptance criteria** will be (very similar to Column Comparise):

1. I should be able to script this component in MDX with as little friction to remember as possible. This means that each child that MDX sees should be the equivalent of 1 Category Box. 

**NOTE: Mind the gaps. MDX is picky about line breaks (similarly to MD) on how it recognizes individual children. Notice the line break at the start and end of the component and in-between children.**

```
<Categories>

   Category 1

   - Category 2
 
   - Category 3

   1. Category 4
   
    ### Category 5

    <Box>
      Category 6
    </Box>

</Categories>
```

A more realistic example would be:

```

<Categories>

<Box>
   
<Icon name="folder" size={3}/>

### Bounties

See what tasks we need help with - there are plenty!

</Box>

<Box>
   
<Icon name="world" size={3}/>

### Translations

Help translate our content into other languages.

</Box>

</Categories>

```

2. No margin top for this component when it comes to spacing around other components. But the individual components do have a margin bottom. 

3. This component has also shown that some of the categories can have statuses. Using the child rule and some good css selectors we can look for the first StatusBanner of it's type and apply absolute positioning to pull this off 

![Screen Shot 2020-06-19 at 8 47 25 AM](https://user-images.githubusercontent.com/9157341/85133877-8724b100-b209-11ea-9ea5-67e1d8ece84d.png)

I would expect MDX code for this to look like:

```
<Categories>
    
    <Box>

        <StatusBanner warning> Funding is currently closed </StatusBanner>

        <Icon name="coffee" size={4}/>

        ### Hackathons

        Organising a hackathon? Apply for a hackathon micro grant
 
     </Box>

</Categories>
```

4. Height differences. Height of each category child should be the same height as it's tallest parent. minHeight is 260px based on spec.

**Comment by MaximumCrash** [comment]
Closed by #68
